### PR TITLE
Submission: Warning if user cancels submission (EXPOSUREAPP-2509)

### PR DESCRIPTION
**Pull request by AndroidMedaGalaxy**
## Implementation screenshot
submission_flow_cancelled

## Description

When a user presses the back navigation button on top or the back navigation button for Android system,
1. Users will get a warning message about cancelling the keys submission flow. They will have 2 options, Cancel and Ok respectively

2. If they choose **Yes** the user starts again from Testergebnis screen

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt [hunk 1]
```diff
@@ -139,10 +149,28 @@ class SubmissionResultPositiveOtherWarningFragment : Fragment(),
...
+    * @see navigateToSubmissionResultFragment
+    */
+    fun handleSubmissionCancellation() {
+        DialogHelper.showDialog(DialogHelper.DialogInstance(
+            requireActivity(),
+            R.string.submission_error_dialog_confirm_cancellation_title,
+            R.string.submission_error_dialog_confirm_cancellation_body,
+            R.string.submission_error_dialog_confirm_cancellation_button_positive,
+            R.string.submission_error_dialog_confirm_cancellation_button_negative,
+            false,
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt:169]: Why is `cancelable=false` here? It means that the user can't exit the dialog with the backbutton, which differs from other dialogs on the same screen.

**Review by d4rken — CHANGES_REQUESTED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt (hunk 1 above)
Comment (ritsam) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt:169]: I'll make that change.

**Review by d4rken — APPROVED** [review]
lgtm

**Review by janetback — APPROVED** [review]
UA approved

**Pull request merged**